### PR TITLE
#30 Migrate to Koin

### PR DESCRIPTION
**Pull request by victorvicari**
- Replaced dagger with koin. 
- a module for every injection needed
- setup a root application class to setup de modules.

### CODE REVIEW

**Comment by victorvicari** [comment]
A nice idea would be have a interface to communicate between activity and presenter. Just like it's done by view. And move the view implementation from the activity to another class. Then, the activity would be pretty much an application controller, without views attached to it.

**Comment by joanb** [comment]
Thank you for your contribution 🎉 
I'll take a look on it as soon as posible.
@all-contributorsbot please add @victorvicari for code

**Comment by joanb** [comment]
@all-contributorsbot please add @victorvicari for code

File: app/src/main/java/com/theopensourcefamily/chessclock/modules/schedulers.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import io.reactivex.android.schedulers.AndroidSchedulers
+import org.koin.dsl.module
+
+val schedulers = module {
+
+  single { provideMainScheduler() }
+
+}
+
+fun provideMainScheduler(): Scheduler? = AndroidSchedulers.mainThread()
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/modules/schedulers.kt]: This function doesn't need to be nullable (actually the presenter is constructed with a non nullable scheduler).
I think we could avoid creating this function until it's needed, and just provide directly the Scheduler:
`  single { AndroidSchedulers.mainThread() } `.
What do you think? 😄

**Review by joanb — COMMENTED** [review]
As we are deleting Dagger completely, we should also remove the Dagger related test coverage exceptions.
This is done by deleting this three lines in the project build gradle:
```
            '**/*_MembersInjector.*',
            '**/*Dagger*Component*.*',
            '**/**Factory*.*'
```

File: app/src/main/java/com/theopensourcefamily/chessclock/modules/schedulers.kt (hunk 1 above)
Comment (victorvicari) [reply app/src/main/java/com/theopensourcefamily/chessclock/modules/schedulers.kt]: Sure. My mistake.

**Comment by joanb** [comment]
@all-contributorsbot please add  @victorvicari for code

**Comment by joanb** [comment]
@all-contributors please add @victorvicari for code

**Comment by joanb** [comment]
LGTM once it passes the build :)
Could you please elaborate on your interface idea? Feel free to create an issue for it, or event a pull request
Thank you for your contribution, you're always welcome here 🎉 
Keep up the good work!

**Comment by joanb** [comment]
Any news on this @victorvicari?, otherwise I'll fix it :)

**Pull request merged**